### PR TITLE
Update docs on Conditions

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
Update docs to refer to upstream implementation for Conditions.

**Motivation for the change:**
Removed status package from operator-lib.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: website/content/en/docs/building-operators/golang/advanced-topics.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ weight: 70
 
 ### Manage CR status conditions
 
-An often-used pattern is to include `Conditions` in the status of custom resources. Conditions represent the latest available observations of an object's state (see the [Kubernetes API conventionsdocumentation][typical-status-properties] for more information).
+An often-used pattern is to include [`Condition`][apimachinery_condition] in the status of custom resources. Conditions represent the latest available observations of an object's state (see the [Kubernetes API conventions documentation][typical-status-properties] for more information).
```
Comment (joelanford) [inline website/content/en/docs/building-operators/golang/advanced-topics.md:9]: Nit for slightly better readability.
```suggestion
An often-used pattern is to include Conditions in the status of custom resources. A [`Condition`][apimachinery_condition] represents the latest available observation of an object's state (see the [Kubernetes API conventions documentation][typical-status-properties] for more information).
```

File: website/content/en/docs/building-operators/golang/advanced-topics.md [hunk 2]
```diff
@@ -19,16 +19,16 @@ To use conditions in your custom resource, add a Conditions field to the Status
...
 ```Go
 import (
-    "github.com/operator-framework/operator-lib/status"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type MyAppStatus struct {
     // Conditions represent the latest available observations of an object's state
-    Conditions status.Conditions `json:"conditions"`
+    Conditions metav1.Condition `json:"conditions"`
```
Comment (joelanford) [inline website/content/en/docs/building-operators/golang/advanced-topics.md:27]:
```suggestion
    Conditions []metav1.Condition `json:"conditions"`
```

**Review by joelanford — APPROVED** [review]
A nit and a minor fix. Other than that,

/lgtm

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by RobEarl** [comment]
Should this be mentioned in the migration guide? https://sdk.operatorframework.io/docs/building-operators/golang/project_migration_guide/

**Comment by varshaprasad96** [comment]
@RobEarl - I suppose this wouldn't be a breaking change as we are not changing any current implementation in SDK and instead directing the users to utilize upstream implementation of `Conditions`. But I am not entirely sure, @joelanford, @jmrodri - can I have your inputs?

**Pull request merged**